### PR TITLE
v1.28 Backports 2024-06-01

### PR DESCRIPTION
**Pull request by sayboras**
* [x] #742 (@jrajahalme)
 * [x] #785 (@jrajahalme)
 * [x] #787 (@sayboras)
 * [x] #778 (@jrajahalme)

Once this PR is merged, a GitHub action will update the labels of these PRs:
```upstream-prs
 742 785 787 778
```

### CODE REVIEW

File: cilium/network_policy.cc [hunk 1]
```diff
@@ -1063,28 +1064,32 @@ class PolicyInstanceImpl : public PolicyInstance {
...
                    [this]() { subscription_->start({}); }),
       transport_factory_context_(
           std::make_shared<Server::Configuration::TransportSocketFactoryContextImpl>(
               context.getServerFactoryContext(),
               context.getTransportSocketFactoryContext().sslContextManager(), *scope_,
               context.getServerFactoryContext().clusterManager(),
-              context.messageValidationContext().dynamicValidationVisitor())) {
+              context.getServerFactoryContext()
+                  .messageValidationContext()
+                  .dynamicValidationVisitor())) {
```
Comment (jrajahalme) [inline cilium/network_policy.cc:1082]: Cleanup, basically replace `context.getServerFactoryContext()` with `context_` wherever possible (+ `make fix`):
```
 // Common base constructor
 // This is used directly for testing with a file-based subscription
 NetworkPolicyMap::NetworkPolicyMap(Server::Configuration::FactoryContext& context)
-    : context_(context.getServerFactoryContext()),
-      tls_map_(context.getServerFactoryContext().threadLocal()),
-      local_ip_str_(
-          context.getServerFactoryContext().localInfo().address()->ip()->addressAsString()),
+    : context_(context.getServerFactoryContext()), tls_map_(context_.threadLocal()),
+      local_ip_str_(context_.localInfo().address()->ip()->addressAsString()),
       name_(fmt::format("cilium.policymap.{}.{}.", local_ip_str_, ++instance_id_)),
-      scope_(context.getServerFactoryContext().serverScope().createScope(name_)),
+      scope_(context_.serverScope().createScope(name_)),
       init_target_(fmt::format("Cilium Network Policy subscription start"),
                    [this]() { subscription_->start({}); }),
       transport_factory_context_(
           std::make_shared<Server::Configuration::TransportSocketFactoryContextImpl>(
-              context.getServerFactoryContext(),
-              context.getTransportSocketFactoryContext().sslContextManager(), *scope_,
-              context.getServerFactoryContext().clusterManager(),
-              context.getServerFactoryContext()
-                  .messageValidationContext()
-                  .dynamicValidationVisitor())) {
+              context_, context.getTransportSocketFactoryContext().sslContextManager(), *scope_,
+              context_.clusterManager(),
+              context_.messageValidationContext().dynamicValidationVisitor())) {
   // Use listener init manager for the first initialization
   transport_factory_context_->setInitManager(context.initManager());
   context.initManager().add(init_target_);
```

File: cilium/network_policy.cc [hunk 2]
```diff
@@ -1063,28 +1064,32 @@ class PolicyInstanceImpl : public PolicyInstance {
...
   context.initManager().add(init_target_);
 
   ENVOY_LOG(trace, "NetworkPolicyMap({}) created.", name_);
   tls_map_.set([&](Event::Dispatcher&) { return std::make_shared<ThreadLocalPolicyMap>(); });
 
-  if (context.admin().has_value()) {
+  if (context_.admin().has_value()) {
     ENVOY_LOG(debug, "Registering NetworkPolicies to config tracker");
-    config_tracker_entry_ = context.admin()->getConfigTracker().add(
+    config_tracker_entry_ = context.getServerFactoryContext().admin()->getConfigTracker().add(
```
Comment (jrajahalme) [inline cilium/network_policy.cc:1092]: same here: `context.getServerFactoryContext()` -> `context_`.

**Review by jrajahalme — CHANGES_REQUESTED** [review]
Small cleanup I'd like to see on the 4th commit.

File: cilium/network_policy.cc (hunk 2 above)
Comment (sayboras) [reply cilium/network_policy.cc:1092]: ah good catch, I changed to `context_` in some places, but forgot to change for all.

File: cilium/network_policy.cc [hunk 3]
```diff
@@ -1063,28 +1063,28 @@ class PolicyInstanceImpl : public PolicyInstance {
 // Common base constructor
 // This is used directly for testing with a file-based subscription
 NetworkPolicyMap::NetworkPolicyMap(Server::Configuration::FactoryContext& context)
-    : tls_map_(context.threadLocal()),
-      local_ip_str_(context.localInfo().address()->ip()->addressAsString()),
+    : context_(context.getServerFactoryContext()),
+      tls_map_(context.getServerFactoryContext().threadLocal()),
```
Comment (jrajahalme) [inline cilium/network_policy.cc:1068]: missed this: -> `tls_map_(context_.threadLocal())`

Comment (sayboras) [reply cilium/network_policy.cc]: fixed and thanks :heavy_check_mark:

**Review by jrajahalme — APPROVED** [review]
Looks great, thanks!

**Pull request merged**